examples/scenes/src/mmark.rs

```diff
@@ -0,0 +1,201 @@
+//! implement the stroke style parameters, and it has not been carefully validated.
+
+use std::cmp::Ordering;
+
+use rand::{seq::SliceRandom, Rng};
+use vello::peniko::Color;
+use vello::{
+    kurbo::{Affine, BezPath, CubicBez, Line, ParamCurve, PathSeg, Point, QuadBez},
+    peniko::Stroke,
+    SceneBuilder,
+};
+
+use crate::{SceneParams, TestScene};
+
+const WIDTH: usize = 1600;
+const HEIGHT: usize = 900;
+
+const GRID_WIDTH: i64 = 80;
+const GRID_HEIGHT: i64 = 40;
+
+pub struct MMark {
+    elements: Vec<Element>,
+}
+
+struct Element {
+    seg: PathSeg,
+    color: Color,
+    width: f64,
+    is_split: bool,
+    grid_point: GridPoint,
+}
+
+#[derive(Clone, Copy)]
+struct GridPoint(i64, i64);
+
+impl MMark {
+    pub fn new(n: usize) -> MMark {
+        let mut result = MMark { elements: vec![] };
+        result.resize(n);
+        result
+    }
+
+    fn resize(&mut self, n: usize) {
+        let old_n = self.elements.len();
+        match n.cmp(&old_n) {
+            Ordering::Less => self.elements.truncate(n),
+            Ordering::Greater => {
+                let mut last = self
+                    .elements
+                    .last()
+                    .map(|e| e.grid_point)
+                    .unwrap_or(GridPoint(GRID_WIDTH / 2, GRID_HEIGHT / 2));
+                self.elements.extend((old_n..n).map(|_| {
+                    let element = Element::new_rand(last);
+                    last = element.grid_point;
+                    element
+                }));
+            }
+            _ => (),
+        }
+    }
+}
+
+impl TestScene for MMark {
+    fn render(&mut self, sb: &mut SceneBuilder, params: &mut SceneParams) {
+        let c = params.complexity;
+        let n = if c < 10 {
+            (c + 1) * 1000
+        } else {
+            ((c - 8) * 10000).min(120_000)
+        };
+        self.resize(n);
+        let mut rng = rand::thread_rng();
+        let mut path = BezPath::new();
+        let len = self.elements.len();
+        for (i, element) in self.elements.iter_mut().enumerate() {
+            if path.is_empty() {
+                path.move_to(element.seg.start());
+            }
+            match element.seg {
+                PathSeg::Line(l) => path.line_to(l.p1),
+                PathSeg::Quad(q) => path.quad_to(q.p1, q.p2),
+                PathSeg::Cubic(c) => path.curve_to(c.p1, c.p2, c.p3),
+            }
+            if element.is_split || i == len {
+                // This gets color and width from the last element, original
+                // gets it from the first, but this should not matter.
+                sb.stroke(
+                    &Stroke::new(element.width as f32),
+                    Affine::IDENTITY,
+                    element.color,
+                    None,
+                    &path,
+                );
+                path.truncate(0); // Should have clear method, to avoid allocations.
```

Comment:
This increases our kurbo requirement (effectively requiring a `cargo update`)

That's probably fine, but it might be worth telling cargo directly about this (that is, we'd currently fail under `-Z minimal-versions`)

Reply:
I'm going to leave this. We get our kurbo dep through peniko, so it's not trivial to express and would need to be reworked later. We'll certainly be bumping our kurbo dep because of the stroke work. But thanks for the heads up!